src/api/app/services/workflows/yaml_downloader.rb

```diff
@@ -15,46 +15,42 @@ def call
-                                                     "\nIs the configuration file in the expected place? Check #{DOCUMENTATION_LINK}\n#{e.message}"
-    end
-
-    def download_url
       # When an external URL is given, it prevails over the path.
-      return @token.workflow_configuration_url if @token.workflow_configuration_url.present?
+      return download_from_url(@token.workflow_configuration_url) if @token.workflow_configuration_url.present?
 
       case @scm_payload[:scm]
-      when 'github'
-        client = Octokit::Client.new(access_token: @token.scm_token, api_endpoint: @scm_payload[:api_endpoint])
-        # :ref can be the name of the commit, branch or tag.
-        client.content("#{@scm_payload[:target_repository_full_name]}", path: "/#{@token.workflow_configuration_path}", ref: @scm_payload[:target_branch])[:download_url]
       when 'gitea'
-        gitea_download_url
+        download_from_url(gitea_configuration_file_url)
+      when 'github'
+        download_github_yaml_file
+      when 'gitlab'
+        download_gitlab_yaml_file
       end
-    rescue Octokit::InvalidRepository => e
-      raise Token::Errors::NonExistentRepository, e.message
-    rescue Octokit::NotFound => e
-      # 'target_branch' can contain a commit sha (when tag push) instead of a branch name
-      raise Token::Errors::NonExistentWorkflowsFile,
-            "#{@token.workflow_configuration_path} could not be downloaded from the SCM branch/commit #{@scm_payload[:target_branch]}: #{e.message}"
     end
 
-    def gitea_download_url
+    def gitea_configuration_file_url
       if @scm_payload[:tag_name].present?
         "#{@scm_payload[:api_endpoint]}/#{@scm_payload[:target_repository_full_name]}/raw/tag/#{@scm_payload[:tag_name]}/#{@token.workflow_configuration_path}"
       else
         "#{@scm_payload[:api_endpoint]}/#{@scm_payload[:target_repository_full_name]}/raw/branch/#{@scm_payload[:target_branch]}/#{@token.workflow_configuration_path}"
       end
     end
 
+    def download_github_yaml_file
+      client = Octokit::Client.new(access_token: @token.scm_token, api_endpoint: @scm_payload[:api_endpoint])
+      # :ref can be the name of the commit, branch or tag.
+      begin
+        content = client.content("#{@scm_payload[:target_repository_full_name]}", path: "/#{@token.workflow_configuration_path}", ref: @scm_payload[:target_branch])[:content]
+      rescue Octokit::InvalidRepository => e
+        raise Token::Errors::NonExistentRepository, e.message
+      rescue Octokit::NotFound => e
+        # 'target_branch' can contain a commit sha (when tag push) instead of a branch name
+        raise Token::Errors::NonExistentWorkflowsFile,
+              "#{@token.workflow_configuration_path} could not be downloaded from the SCM branch/commit #{@scm_payload[:target_branch]}: #{e.message}"
+      end
+      create_temp_file(Base64.decode64(content))
```

Comment:
Hm, why do we need to create a temp file?

Reply:
This has already been discussed. It's part of #14750